### PR TITLE
Add notification bar tests

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Added tests for primaryBar, primaryOutlineBar, and neutralBar

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| 7 - Title | 8 - Bar |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Notification/src/Notification.types.ts [hunk 1]
```diff
@@ -3,12 +3,13 @@ import { IViewProps, ITextProps } from '@fluentui-react-native/adapters';
...
 import { FontTokens, IBorderTokens, IColorTokens, LayoutTokens } from '@fluentui-react-native/tokens';
 import { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+import { FlexStyle } from 'react-native';
 
 export const notification = 'Notification';
 export const NotificationVariants = ['primary', 'neutral', 'primaryBar', 'primaryOutlineBar', 'neutralBar', 'danger', 'warning'] as const;
 export type NotificationVariant = typeof NotificationVariants[number];
 
-export interface NotificationTokens extends LayoutTokens, IBorderTokens, IColorTokens, FontTokens {
+export interface NotificationTokens extends LayoutTokens, IBorderTokens, IColorTokens, FontTokens, FlexStyle {
```
Comment (Saadnajmi) [inline packages/components/Notification/src/Notification.types.ts:12]: Why did you need to extend FlexStyle? I'm guessing `LayoutTokens` didn't have what you wanted?

File: packages/components/Notification/src/NotificationTokens.ts [hunk 2]
```diff
@@ -4,41 +4,61 @@ import { NotificationTokens } from './Notification.types';
...
 export const defaultNotificationTokens: TokenSettings<NotificationTokens, Theme> = (t: Theme) =>
   ({
+    alignSelf: 'flex-start',
     backgroundColor: t.colors.background,
+    borderColor: 'transparent',
+    borderRadius: 12,
+    borderWidth: 1,
     color: t.colors.brandForeground1,
     fontFamily: 'primary',
+    fontLetterSpacing: -0.24,
```
Comment (Saadnajmi) [inline packages/components/Notification/src/NotificationTokens.ts:14]: Because this is iOS specific, it might be good to mark so with a comment " //iOS only prop" at the end. That way, if/when we bring Notification to other platforms like android, it's easier for us to know what to move around

File: packages/components/Notification/src/Notification.types.ts (hunk 1 above)
Comment (joannaquu) [reply packages/components/Notification/src/Notification.types.ts:12]: Yeah, LayoutTokens didn't include alignSelf

File: packages/components/Notification/src/NotificationTokens.ts [hunk 3]
```diff
@@ -4,41 +4,61 @@ import { NotificationTokens } from './Notification.types';
...
+      fontLetterSpacing: -0.08,
+      fontLineHeight: 18,
       fontSize: 13,
       fontWeight: '400',
-      fontLineHeight: 18,
-      fontLetterSpacing: -0.08,
+      paddingVertical: 12,
+    },
+    isBar: {
+      alignSelf: 'center',
```
Comment (Saadnajmi) [inline packages/components/Notification/src/NotificationTokens.ts:28]: Any chance we can do this in Notification.styling.ts and not make it a token?

File: packages/components/Notification/src/Notification.tsx [hunk 4]
```diff
@@ -34,17 +34,30 @@ export const Notification = compose<NotificationType>({
...
+      marginHorizontal = 0;
+      alignSelf = 'center' as FlexAlignType;
+    }
+
+    let rootStyle = useMemo(() => {
+      return { variant: userProps.variant, marginHorizontal: marginHorizontal };
+    }, ['variant', 'marginHorizontal']);
+    let messageStyle = useMemo(() => {
+      return { alignSelf: alignSelf };
+    }, ['alignSelf']);
```
Comment (Saadnajmi) [inline packages/components/Notification/src/Notification.tsx:51]: So that we have more const variables,

```suggestion
    const isBar = (['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant))
    const marginHorizontal = isBar ? 0: 16
    const alignSelf = isBar ? 'center' : 'flex-start';

    const rootStyle: ViewStyle = useMemo(() => {
      return { style: { marginHorizontal: marginHorizontal } };
    }, ['variant', 'marginHorizontal']);
    let messageStyle: ViewStyle = useMemo(() => {
      return { style: {alignSelf: alignSelf } };
    }, ['alignSelf']);
```

Comment (Saadnajmi) [reply packages/components/Notification/src/Notification.tsx:51]: Or better yet:

```suggestion
    const isBar = (['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant))
    const rootStyle: ViewStyle = useMemo(() => {
      const marginHorizontal = isBar ? 0: 16
      return { style: { marginHorizontal: marginHorizontal } };
    }, ['isBar', ]);
    let messageStyle: ViewStyle = useMemo(() => {
      const alignSelf = isBar ? 'center' : 'flex-start';
      return { style: {alignSelf: alignSelf } };
    }, ['isBar']);
```

Comment (joannaquu) [reply packages/components/Notification/src/Notification.tsx]: Had to change a couple of things (variant has to be in rootStyle/rootStyle can't be of type ViewStyle for mergeProps to work) to resolve errors so lmk if there's a change here you want to keep

File: packages/components/Notification/src/Notification.tsx [hunk 5]
```diff
@@ -30,15 +35,24 @@ export const Notification = compose<NotificationType>({
   },
   useRender: (userProps: NotificationProps, useSlots: UseSlots<NotificationType>) => {
     const Slots = useSlots(userProps, (layer) => notificationLookup(layer, userProps));
+    const isBar = ['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant);
+    const rootStyle = useMemo(() => {
+      const marginHorizontal = isBar ? 0 : 16;
+      return { variant: userProps.variant, marginHorizontal: marginHorizontal };
+    }, ['variant', 'isBar']);
```
Comment (Saadnajmi) [inline packages/components/Notification/src/Notification.tsx:42]: `variant` is not a styeProp. The final object returned is a mix of one Notification prop, and one style prop. We're also not modifying variant at all, so I'd rather it not be in the returned object. This code as written would not place "marginHorizontal" inside Slots.root's style I believe. 

To enforce type checking so that VSCode's intelligence can check this error, we can add some type annotations:

```suggestion
    const rootStyle: ViewStyle = useMemo(() => {
      const marginHorizontal = isBar ? 0 : 16;
      return { marginHorizontal: marginHorizontal };
    }, [ 'isBar']);
```

File: packages/components/Notification/src/Notification.tsx (hunk 4 above)
Comment (Saadnajmi) [reply packages/components/Notification/src/Notification.tsx]: Resolved offline

File: packages/components/Notification/src/Notification.tsx (hunk 5 above)
Comment (Saadnajmi) [reply packages/components/Notification/src/Notification.tsx]: resolved offline

File: apps/fluent-tester/src/FluentTester/TestComponents/Notification/NotificationTest.tsx [hunk 6]
```diff
@@ -77,6 +77,7 @@ const PrimaryBarTest: React.FunctionComponent = () => {
   return (
     <Notification
       variant={'primaryBar'}
+      title="Kat's iPhone X"
```
Comment (Saadnajmi) [inline apps/fluent-tester/src/FluentTester/TestComponents/Notification/NotificationTest.tsx:80]: Undo? Though it's fine to keep too :)

File: packages/components/Notification/src/Notification.styling.ts [hunk 7]
```diff
@@ -70,21 +70,17 @@ export const stylingSettings: UseStylingOptions<NotificationProps, NotificationS
...
+        style: {
+          alignSelf: 'center',
+          marginLeft: 16,
+        },
+        appearance: 'subtle',
+        minWidth: 0,
+        padding: 0,
+        paddingHorizontal: 0,
+      };
+    }, ['color']),
```
Comment (Saadnajmi) [inline packages/components/Notification/src/Notification.styling.ts:84]: No longer need the 'color' key

**Review by Saadnajmi — APPROVED** [review]

File: packages/components/Notification/src/NotificationTokens.ts [hunk 8]
```diff
@@ -5,40 +5,56 @@ import { NotificationTokens } from './Notification.types';
...
       color: '#BC2F34',
-      borderRadius: 12,
     },
     warning: {
       backgroundColor: '#FFFBD6',
       color: '#4C4400',
-      borderRadius: 12,
+    },
+    primaryBar: {
+      backgroundColor: '#EBF3FC',
```
Comment (ValentinaKozlova) [inline packages/components/Notification/src/NotificationTokens.ts:46]: Do you have these colors in globalTokens or theme colors?
